src/modules/fw_autotune_attitude_control/fw_autotune_attitude_control.cpp

```diff
@@ -620,21 +620,51 @@ void FwAutotuneAttitudeControl::saveGainsToParams()
 
 const Vector3f FwAutotuneAttitudeControl::getIdentificationSignal()
 {
-	if (_steps_counter > _max_steps) {
-		_signal_sign = (_signal_sign == 1) ? 0 : 1;
-		_steps_counter = 0;
 
-		if (_max_steps > 1) {
-			_max_steps--;
 
-		} else {
-			_max_steps = 5;
+	const hrt_abstime now = hrt_absolute_time();
+	const float t = static_cast<float>(now - _state_start_time) * 1e-6f;
+	float signal = 0.0f;
+
+	switch (_param_fw_sysid_signal_type.get()) {
+	case  static_cast<int32_t>(SignalType::kStep): {
+			if (_steps_counter > _max_steps) {
+				_signal_sign = (_signal_sign == 1) ? 0 : 1;
+				_steps_counter = 0;
+
+				if (_max_steps > 1) {
+					_max_steps--;
+
+				} else {
+					_max_steps = 5;
+				}
+			}
+
+			signal = float(_signal_sign) * _param_fw_at_sysid_amp.get();
 		}
+		break;
+
+	case static_cast<int32_t>(SignalType::kLinearSineSweep): {
+
+			signal = signal_generator::getLinearSineSweep(_param_fw_at_sysid_f0.get(),
+					_param_fw_at_sysid_f1.get(),
+					_param_fw_sysid_time.get(), t);
+		}
+		break;
+
+	case static_cast<int32_t>(SignalType::kLogSineSweep): {
+			signal = signal_generator::getLogSineSweep(_param_fw_at_sysid_f0.get(), _param_fw_at_sysid_f1.get(),
+					_param_fw_sysid_time.get(), t);
+		}
+		break;
+
+	default:
+		signal = 0.f;
+		break;
 	}
 
 	_steps_counter++;
```

Comment:
That's quite specific to the step signal, please move it in the corresponding `if`

Reply:
which if condition I can move

Reply:
you can move it there: https://github.com/PX4/PX4-Autopilot/pull/22666/files/6f307029cfc0a63f4a517a630b3879634139242e#diff-84c0821267981c4b32c33723bb8a81656d2f429ade74188096c065e38ba64d70R642